dataland-frontend/src/components/forms/parts/UploadReports.vue

```diff
@@ -60,49 +64,76 @@
 <script lang="ts">
 import { defineComponent } from "vue";
 import PrimeButton from "primevue/button";
-import { CompanyReport } from "@clients/backend";
 import ReportFormElement from "@/components/forms/parts/ReportFormElement.vue";
 import ElementsDialog from "@/components/general/ElementsDialog.vue";
 import { ReportToUpload, StoredReport } from "@/utils/FileUploadUtils";
 import UploadDocumentsForm from "@/components/forms/parts/elements/basic/UploadDocumentsForm.vue";
+import { CompanyReport } from "@clients/backend";
+import { ObjectType } from "@/utils/UpdateObjectUtils";
 
 export default defineComponent({
   name: "UploadReports",
+  inheritAttrs: false,
+  inject: {
+    injectReferencedReportsForPrefill: {
+      from: "referencedReportsForPrefill",
+      default: {},
+    },
+  },
   components: {
     UploadDocumentsForm,
     ReportFormElement,
     PrimeButton,
   },
-  emits: ["referenceableReportNamesChanged"],
+  emits: ["reportsUpdated"],
   data() {
     return {
       reportsToUpload: [] as ReportToUpload[] | undefined,
       storedReports: [] as StoredReport[],
     };
   },
   props: {
+    name: {
+      type: String,
+      required: true,
+    },
     referencedReportsForPrefill: {
       type: Object as () => { [key: string]: CompanyReport },
     },
-  },
-  computed: {
-    allReferenceableReportNames(): string[] {
-      const namesOfFilesToUpload = this.reportsToUpload.map((reportToUpload) => reportToUpload.fileNameWithoutSuffix);
-      const namesOfStoredReports = this.storedReports.map((storedReport) => storedReport.reportName);
-      return namesOfFilesToUpload.concat(namesOfStoredReports);
+    isEuTaxonomy: {
+      type: Boolean,
+      default: false,
     },
   },
   watch: {
     referencedReportsForPrefill() {
       this.prefillAlreadyUploadedReports();
     },
   },
+  computed: {
+    namesOfReportsToUpload() {
+      return this.reportsToUpload.map((reportToUpload) => reportToUpload.fileNameWithoutSuffix);
```

Comment:
You run ".map()" on a possibly undefined variable here.   That can lead to a runtime error I think.
Can we remove `undefined` as possible type of `reportsToUpload` ?  Or will this break anything?

Alternative is to check if it is defined, before running this .map() code.

Reply:
Looking at it

Reply:
Waiting for approval from the other devs, that the removal of this `undefined` was ok.

Reply:
> Removed the undefined. Or was there a good reason that we needed this? @MarcinStanczyk7 @pioSko2

removing undefined is ok

Reply:
👍